src/compiler/types.ts

```diff
@@ -10151,7 +10170,78 @@ export type PragmaArgumentType<KPrag extends keyof ConcretePragmaSpecs> = Concre
     : never;
 
 /** @internal */
-export type ConcretePragmaSpecs = typeof commentPragmas;
+export interface ConcretePragmaSpecs {

```

Comment:
🥹

This here is why you should only enable `isolatedDeclarations` if you _really need to_, and definitely not as a default - this type is automatic and you don't have to think about it normally, since it's just derived from the expressions, but once you're forced to write it out, it's huge (despite its' simplicity!). The worst part is that now the expressions in the `commentPragmas` const and this type can accidentally diverge subtly, since we don't have "exact types". It is a degradation in UX, _and_ in type safety. Very unfortunate, but so long as we want to self-test `isolatedDeclarations`, there's no avoiding it. 🙍

Reply:
Actually, since `commentPragmas` is already an `as const` tree of exclusively literal expressions, why does it even have an ID error? The types should be trivially inferrable.

Reply:
Yeah, I think that we should find a way to derive this from some other type.

Reply:
It doesn't know what to do with the `PragmaKindFlags`